src/customTypes.ts

```diff
@@ -0,0 +1,96 @@
+import type { Events, Message } from "discord.js";
+
+export type Command = {
+  name: string;
+  command: string | RegExp;
+  description: string;
+  showInHelp: boolean;
+  match: (message: Message) => boolean;
+  execute: (message: Message) => MaybePromiseVoid;
+};
+
+export const CommandGuard = (object: object) =>
+  "match" in object
+  && "execute" in object;
```

Comment:
```suggestion
export const isBotCommand = (object: object) => "match" in object && "execute" in object;
```

Use consistent naming, according to your deno linting rules.